core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -38,7 +38,13 @@ object ConsoleConsumer extends Logging {
 
   def main(args: Array[String]) {
     val conf = new ConsumerConfig(args)
-    run(conf)
+    try {
+      run(conf)
+    } catch {
+      case e: Throwable =>
+        error("Error running consumer: ", e)
+    }
+    System.exit(0);
```

Comment:
Why is this needed? Are we leaking non-daemon threads? Also, we should probably exit with `System.exit(-1)` when there is an error.